app/scripts/directives/overview/virtualMachineRow.js

```diff
@@ -24,6 +24,76 @@
     templateUrl: 'views/overview/_virtual-machine-row.html'
   });
 
+  var RDP_PORT = 3389;
+
+  function fileDownload(data, fileName, mimeType) {
+    fileName = fileName || "vm.rdp";
+    mimeType = mimeType || "application/rdp";
+
+    var a = document.createElement('a');
+    if (navigator.msSaveBlob) { // IE10
+      return navigator.msSaveBlob(new Blob([data], { type: mimeType }), fileName);
+    } else if ('download' in a) { // html5 A[download]
+      a.href = 'data:' + mimeType + ',' + encodeURIComponent(data);
+      a.setAttribute('download', fileName);
+      document.body.appendChild(a);
+      a.click();
+      document.body.removeChild(a);
+      return true;
+    } else { // do iframe dataURL download (old ch+FF):
+      var f = document.createElement('iframe');
+      document.body.appendChild(f);
+      f.src = 'data:' + mimeType + ',' + encodeURIComponent(data);
+      setTimeout(function () {
+        document.body.removeChild(f);
+      }, 333);
+      return true;
+    }
+  }
+
+  function buildRdp(address, port) {
```

Comment:
`_.template` might be helpful here.

https://lodash.com/docs/#template

I'd suggest making this a method in an RDPService. The string is large enough that it would be nice to remove it from the controller.

It's a shame we can't use ES6 template literals :(

Reply:
Yes, the ES6 brings a lot of useful stuff.

I moved the code to new RdpService but keeping the string as it is since it's used just once and the template would still need to be provided somewhere. I can change it if needed, but I do not see benefit in this particular case.